sched/adaptdl_sched/supervisor.py

```diff
@@ -66,6 +66,45 @@ async def _handle_discover(self, request):
                         return web.json_response(pod_ip_list)
         return web.json_response(status=408)  # Timeout.
 
+    async def _handle_discover_gpu(self, request):
```

Comment:
This can be implemented as an extension of the existing `discover` endpoint, by making it accept a `jsonpath=` query parameter which defaults to `".status.podIP"`. See https://pypi.org/project/jsonpath-ng/